core/src/main/scala/tofu/package.scala

```diff
@@ -3,13 +3,15 @@ import cats.{ApplicativeError, MonadError}
 import cats.data.Ior
 
 package object tofu {
+  type In[C, F[_]] = WithContext[F, C]
+
   type HasContext[F[_], C] = Context[F] { type Ctx = C }
 
   type HasLocal[F[_], C] = Local[F] { type Ctx = C }
 
   type HasProvide[F[_], G[_], C] = Provide[F] { type Ctx = C; type Lower[A] = G[A] }
 
-  type HasContextRun[F[_], G[_], C] = RunContext[F] { type Ctx = C; type Lower[A] = G[A] }
+  type HasContextRun[F[_], G[_], C] = RunContext[F] { type Lower[A] = G[A]; type Ctx = C; }
```

Comment:
Does this swap affect inference anyhow? Why hasn't HasProvide been changed?

Reply:
No, this change is insignificant. Just a scar of searching how to make `ReaderT` instance work inside `ContextBase` companion